### PR TITLE
[Enhancement] Make optimizer customizable

### LINKED ISSUE #12851: Make optimizer customizable for materialized view rewrite
## Feature request

**Is your feature request related to a problem? Please describe.**

For materialized view rewrite, we first compile the mv's plan by only rule-based rewrite. But now the optimizer can not be used by only rule-based rewrite. So I want to make the optimizer more customizable, which include:
1. can choose whether to use rule-based optimization or cost-based optimization
2.  can customize whether to enable an rule or not

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by ABingHuang**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #12851 

## Problem Summary(Required) ：

Make optimizer customizable, which will be used in materialized view rewrite optimization.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/OptimizerConfig.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    public enum OptimizerAlgorithm {
+        RULE_BASED,
+        COST_BASED
+    }
+    private OptimizerAlgorithm optimizerAlgorithm;
+
+    private BitSet ruleSetSwitches;
+    private BitSet ruleSwitches;
+
+    private static OptimizerConfig DEFAULT_CONFIG = new OptimizerConfig();
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/OptimizerConfig.java:20]:
```suggestion
    private static final OptimizerConfig DEFAULT_CONFIG = new OptimizerConfig();
```

Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/OptimizerConfig.java]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 65 / 66 (98.48%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/task/OptimizerTask.java|2|3|66.67%|[47]|
|:large_blue_circle:|com/starrocks/sql/optimizer/Optimizer.java|35|35|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/OptimizerContext.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/OptimizerConfig.java|21|21|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/RuleSetType.java|1|1|100.00%|[]|

**Comment by ABingHuang** [comment]
run starrocks_admit_test

**Review by Seaven — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**